validator/node/node.go

```diff
@@ -149,6 +159,28 @@ func (c *ValidatorClient) Start() {
 		panic("Panic closing the validator client")
 	}()
 
+	// If --enable-duplicate-validator-detection flag is set be ready for an event feed.
+	if c.duplicateCheckFlag {
+		go func() {
```

Comment:
This looks like a dubious addition to the general node handling code. I am not sure why there is required, it basically sidesteps where we generally handling node shutdowns by running a separate exit handling routine.

Reply:
I thought that was a smart way of shutting down.  Do you have examples on how its done otherwise?

Reply:
Ideally we should have only one place where a node can be shutdown, having it for one specific feature doesn't seem like the best approach. Currently our service registry structure suggests that for abrupt shut downs we would use fatal logs. We can continue using fatal logs here, it *should* be fine. For the most part, this check is done at startup so any abrupt shutdowns dont have any adverse effects on the consistency of data persisted in our db. 

However you do bring up a good point on what would be a good way to shut down a node correctly internally( without sigInt or sigkill) . We do not have the feature currently  for either the beacon node/validator. This would be worth addressing in another issue/PR. And once it is done there, we can use that to correctly shut down the validator(rather than a fatal log).